src/AutomatedCar/Models/World.cs

```diff
@@ -22,9 +21,51 @@ public class World : ReactiveObject {
         public int Width { get; set; }
         public int Height { get; set; }
 
-        public World () { }
-        public void addObject (WorldObject worldObject) {
+        private World () { }
+        public void addObject (IWorldObject worldObject) {
             WorldObjects.Add (worldObject);
         }
+
+        public List<IWorldObject> SearchInRange(List<Point> points)
```

Comment:
Igen, erre írtad, hogy kell valami megoldás. Úgy is lesz mindennek polygonja, olyan lib-et kellene találni, amelynek elég csak odaadni a poligont, legrosszabb esetben a poligon pontlistáját (ha nem kompatibilis azzal az osztállyal ami a megjelenítésnél az avaloniában van) és egy intersects metódussal eldönti a kérdést.

Reply:
Én erre találtam most egy ilyet, ránézésre jónak tűnik mindjárt kipróbálom:
https://sourceforge.net/projects/polyclipping/
Nuget-ben: 'Clipper'